### PR TITLE
Remove incorrent documentation about finalizer tasks

### LINKED ISSUE #13357: finalizer tasks are not up to date when finalized tasks are up to date
https://discuss.gradle.org/t/finalizedby-tasks-runs-even-though-the-original-task-is-up-to-date/29150

From the documentation:

> On the other hand, finalizer tasks are not executed if the finalized task didn’t do any work, for example if it is considered up to date or if a dependent task fails.

This test fails…

```
    @Unroll
    void 'finalizer tasks are up to date if all finalized tasks are up to date'() {
        when:
        buildFile << """
            class NotParallel extends DefaultTask {}
            task a {
                finalizedBy 'f'
                inputs.file("in")
                outputs.file("out")
                doLast {
                    file("out").text = file("in").text
                }
            }
            task f {
                doLast {}
            }
        """
        file("in") << "foo"

        then:
        succeeds("a")
        executedAndNotSkipped(":a", ":f")

        and:
        succeeds("a")
        skipped(":a", ":f")
    }
```

For the second invocation, `:f` is not up to date.

### Expected Behavior

As stated by the documentation. A finalizer should be up to date if all tasks being finalized were up to date.

### Current Behavior

Finalizers are always executed.

### PR DESCRIPTION
**Pull request by asodja**
Fixes #13357

### CODE REVIEW

File: subprojects/docs/src/docs/userguide/authoring-builds/more_about_tasks.adoc [hunk 1]
```diff
@@ -1368,8 +1368,6 @@ include::{snippetsPath}/tasks/finalizersWithFailure/tests-groovy/taskFinalizersW
 ----
 ====
 
-On the other hand, finalizer tasks are not executed if the finalized task didn't do any work, for example if it is considered up to date or if a dependent task fails.
```
Comment (wolfs) [inline subprojects/docs/src/docs/userguide/authoring-builds/more_about_tasks.adoc:1371]: Finalizer tasks are not executed if the finalized tasks did not run - for example when a dependent task of the finalized task failed. I wonder if we need to spell this out here or not. I suppose deleting that sentence is fine.

**Review by wolfs — APPROVED** [review]
LGTM!

File: subprojects/docs/src/docs/userguide/authoring-builds/more_about_tasks.adoc (hunk 1 above)
Comment (asodja) [reply subprojects/docs/src/docs/userguide/authoring-builds/more_about_tasks.adoc:1371]: Ok, I will merge this then.

**Comment by asodja** [comment]
@bot-gradle test and merge

**Pull request merged**